utils/swift_build_sdk_interfaces.py

```diff
@@ -419,6 +419,14 @@ def main():
         # sem_open, so switch to a ThreadPool instead.
         import threading
         shared_output_lock = threading.Lock()
+        # This is a hack, that was added for newer Python versions to again
+        # work around the lack of sem_open.
+        def thread_lock(self):
+            import threading
+            return threading.Lock()
+        ctx = multiprocessing.get_context()
+        ctxType = type(ctx)
+        ctxType.Lock = thread_lock
```

Comment:
[I added the surrounding code to have Android use a ThreadPool instead a couple years ago](https://github.com/apple/swift/pull/27167/files#diff-8da7eca5f7cd5b1d30b731e9c79221c2cb47ac645f1be3dbe5e131c01faf74b9R410) but that also broke with a recent Python 3 version, so I rummaged around in ThreadPool internals till I came up with this hack to get it to run.